### PR TITLE
chore: tree sitter updates

### PR DESCRIPTION
**Pull request by jeremiah-snee-openx**
Download Project & run tree-sitter generate

- Tree sitter auto runs the following changes

```log
Replacing nan dependency with node-addon-api in package.json
Adding node-gyp-build dependency to package.json
Adding prebuildify devDependency to package.json
Adding an install script to package.json
Adding a prebuildify script to package.json
Adding peerDependencies to package.json
Adding types to package.json
Adding files to package.json
Replacing index.js with new binding API
Replacing binding.cc with new binding API
Replacing binding.gyp with new binding API
```

## Summary by CodeRabbit

- **New Features**
    - Introduced a language parser library for Tmux configurations, `tree-sitter-tmux`.
    - Added support for multiple programming languages (Swift, Go, Python, Node.js) to interact with the Tmux grammar.
- **Documentation**
    - Added coding style rules and line ending specifications for better code consistency.
- **Refactor**
    - Refactored Node.js bindings to use `Napi` for improved integration.
- **Tests**
    - Implemented tests for the Go bindings to ensure grammar loading functionality.
- **Chores**
    - Configured build and installation processes for the new parser library.

### CODE REVIEW

**Comment by jeremiah-snee-openx** [comment]
@Freed-Wu Not sure what you want to do for these PR checkers. 

Every change in this commit came from tree-sitter CLI. 

Do you want to add files to .gitignore, or update [.pre-commit-config.yaml](https://github.com/Freed-Wu/tree-sitter-tmux/blob/main/.pre-commit-config.yaml) to ignore certain files & errors

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-exclude: ^(templates|src)/.*
```
Comment (jeremiah-snee-openx) [inline .pre-commit-config.yaml:2]: excluding generated files

File: binding.gyp
Comment (jeremiah-snee-openx) [inline binding.gyp:1]: could add this file to and not manually have to add `# noqa: `
https://github.com/Freed-Wu/tree-sitter-tmux/blob/9061056d04ae76ec958e5e9ae59f6885128d0160/.pre-commit-config.yaml#L2

**Comment by Freed-Wu** [comment]
This large PR is also needed to be splitted to many small PRs, IMO.

**Comment by jeremiah-snee-openx** [comment]
> This large PR is also needed to be splitted to many small PRs, IMO.

How would you like this PR split up? Almost all of the changes are created by the `tree-sitter` cli. The only other changes I have made in the PR are fixing broken PR Checkers.

**Comment by Freed-Wu** [comment]
> How would you like this PR split up

No serious, I just want:

- a commit to add set-window-option
- a commit to update .pre-commit-config.yaml
- a commit to update test/corpus

I have done this work in #4

**Comment by Freed-Wu** [comment]
Thanks for you help!

**Pull request closed without merging**